draft-ietf-quic-transport.md

```diff
@@ -550,37 +550,45 @@ of protection against off-path attackers.
-
-When an an Initial packet is sent by a client which has not previously received
-a Retry packet from the server, it populates the Destination Connection ID field
-with a randomly selected value.  This MUST be at least 8 octets in length. Until
-a packet is received from the server, the client MUST use the same random value
-unless it also changes the Source Connection ID (which effectively starts a new
-connection attempt).  The randomized Destination Connection ID is used to
-determine packet protection keys.
-
-A server sends its first Initial packet in response to a Client Initial.  A
-server may send multiple Initial packets.  The Cryptographic key exchange could
-require multiple round trips or we might perform retransmissions of this data.
-
-The Destination Connection ID field in the server's Initial packet contains a
-connection ID that is chosen by the recipient of the packet; the Source
-Connection ID includes the connection ID that the sender of the packet wishes to
-use (see {{connection-id}}).
-
-On receiving the first Server Initial packet the client uses the Source
-Connection ID supplied by the server as the Destination Connection ID for
-subsequent packets.
-
-If the client received a Retry packet from the server and is sending a second
-Initial packet, then it sets the Destination Connection ID to the value from the
-Source Connection ID in the Retry packet.
-
-Changing Destination Connection ID also results in a change to the keys used to
-protect the Initial packet.
+first CRYPTO_HS frames sent by the client and the server.
+
+The Initial packet has two additional header fields that follow the normal Long
+Header.
+
+~~~
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|        Token Length (i)  ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                            Token (*)                        ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~
+
+Token Length:
+
+: A variable-length integer specifying the length of the Token field, in bytes.
+It may be zero if no token is present. The server MUST send a zero-length
+token.
+
+Token:
+
+: An optional token blob previously received in either a Retry packet or
+NEW_TOKEN frame.
+
+If the client has not previously received a Retry packet from the server, it
+populates the Destination Connection ID field with a randomly selected value.
+This MUST be at least 8 octets in length.  Until a packet is received from the
+server, the client MUST use the same random value unless it also changes the
+Source Connection ID (which effectively starts a new connection attempt).  The
+randomized Destination Connection ID is used to determine packet protection
+keys, but is not included in server packets.
+
+If the client received a Retry packet and is sending a second Initial packet,
```

Comment:
How about "and sends an Initial packet in response,"  Because it may not be the second Initial packet sent.